File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> answer = arr;
    
    if(answer.size() == 1){
        answer[0] = -1;
    }else{
        answer.erase(min_element(answer.begin(), answer.end()));
    }
    return answer;
}

File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        
        // 길이가 1일 때
        if(arr.length == 1)
            return new int[]{-1};
        
        // 최솟값 찾기
        int minVal = Arrays.stream(arr).min().getAsInt();
        
        return Arrays.stream(arr)
            .filter(i -> i != minVal)
            .toArray();
        
    }
}